python/lsst/ts/observatory/control/auxtel/atcalsys.py

```diff
@@ -0,0 +1,421 @@
+
+    Additional usages definition:
+
+    * Setup: Enable ATWhitelight and perform warmup. Enable
+             ATMonochromator, Electrometer, and FiberSpectrographs.
+    * Configure: Adjust ATWhitelight and ATMonochromator for type of flat.
+    * TakeFlat: Take flatfield image with LATISS, Electrometer and FiberSpec
+    * DryTest: Disable CSCs and unit tests
+    """
+
+    Setup = 1 << 3
+    Configure = 1 << 4
+    TakeFlat = 1 << 5
+
+    def __iter__(self) -> typing.Iterator[int]:
+        return iter(
+            [
+                self.All,
+                self.StateTransition,
+                self.MonitorState,
+                self.MonitorHeartBeat,
+                self.Setup,
+                self.Configure,
+                self.TakeFlat,
+                self.DryTest,
+            ]
+        )
+
+
+class ATCalsys(BaseCalsys):
+    """LSST Auxiliary Telescope Calibration System.
+
+    ATCalsys encapsulates core functionality from the following CSCs: ATCamera,
+    ATWhiteLight, ATMonochromoter, Fiberspectrograph, Electrometer.
+
+    Parameters
+    ----------
+    domain : `lsst.ts.salobj.Domain`
+        Domain for remotes. If `None` create a domain.
+    log : `logging.Logger`
+        Optional logging class to be used for logging operations. If `None`,
+        creates a new logger. Useful to use in salobj.BaseScript and allow
+        logging in the class use the script logging.
+    intended_usage: `int`
+        Optional integer that maps to a list of intended operations. This is
+        used to limit the resources allocated by the class by gathering some
+        knowledge about the usage intention. By default allocates all
+        resources.
+    """
+
+    def __init__(
+        self,
+        domain: typing.Optional[salobj.Domain] = None,
+        log: typing.Optional[logging.Logger] = None,
+        intended_usage: typing.Optional[int] = None,
+    ) -> None:
+
+        self.electrometer_index = 1
+        self.fiberspectrograph_index = 1
+
+        super().__init__(
+            components=[
+                "ATCamera",  # isn't this just LATISS?
+                "ATMonochromator",
+                "ATWhiteLight",
+                f"FiberSpectrograph:{self.fiberspectrograph_index}",
+                f"Electrometer:{self.electrometer_index}",
+            ],
+            instrument_setup_attributes=["filter", "calib_type"],
+            domain=domain,
+            log=log,
+            intended_usage=intended_usage,
+        )
+
+        self.entrance_slit = 0.5
+        self.exit_slit = 0.5
+
+        # White lamp config
+        self.timeout_lamp_warm_up = 60 * 15
+        self.track_lamp_warmup = 60
+        self.cmd_timeout = 30
+        self.timeout_std = 20
+        self.whitelight_startup_power = 1000
+
+        # Chiller config
+        self.timeout_chiller_cool_down = 60 * 15
+        self.chiller_temp_tolerance_relative = 0.2
+        self.chiller_temperature = 20
+
+        self.filter_gratings = {
```

Comment:
What are these two dictionaries intended for?

If they are what I think they are, I think you are trying to store calibration settings for different instrument configurations. 

One of the discussions I had with Patrick a long time ago was about how to store these in a way that would be easy to update. I think we decided that the best thing to do was to have these in configuration files that the class would read at runtime. This way, we can expand and update those as needed without the need to change the code.

Reply:
Ok. I have included all of these in the `instrument_setup_attributes` which can be saved in configuration files.